model/src/pyrenew/metaclass.py

```diff
@@ -412,3 +412,76 @@ def plot_posterior(
             draws_col=draws_col,
             obs_col=obs_col,
         )
+
+    def posterior_predictive(
+        self,
+        rng_key: ArrayLike | None = None,
+        *args,
+        guide: Callable | None = None,
+        params: dict | None = None,
+        num_samples: int | None = None,
+        return_sites: list[str] | None = None,
+        infer_discrete: bool = False,
+        parallel: bool = False,
+        batch_ndims: int | None = None,
+        exclude_deterministic: bool = True,
+        **kwargs,
+    ) -> dict:
+        if self.mcmc is None:
+            raise ValueError(
+                "No posterior samples available. Run model with model.run()."
+            )
+
+        if rng_key is None:
+            rand_int = np.random.randint(
+                np.iinfo(np.int64).min, np.iinfo(np.int64).max
+            )
+            rng_key = jr.key(rand_int)
+
+        return Predictive(
+            model=self.sample,
+            posterior_samples=self.mcmc.get_samples(),
+            *args,
+            guide=guide,
+            params=params,
+            num_samples=num_samples,
+            return_sites=return_sites,
+            infer_discrete=infer_discrete,
+            parallel=parallel,
+            batch_ndims=batch_ndims,
+            exclude_deterministic=exclude_deterministic,
+        )(rng_key, **kwargs)
```

Comment:
This was mentioned in Teams. 

> Did you have an (eventual) solution in mind beyond kwargs? I feel it works well for what we have now but might change in the future if there is something better. 

Response:

> It would be nice if you didn’t have to pass any arguments at all. I would want to somehow differentiate between a model and a model with data.